### ISSUE TITLE
resource/aws_appsync_resolver: Add CachingConfig support

### ISSUE THREAD

**Issue by ackintosh** [terraform-providers/terraform-provider-aws#13230]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The `CachingConfig` option is missing in `aws_appsync_resolver` resource. In the AWS console, this is done by toggling `Enable Caching` flag.

The [AWS-CLI](https://docs.aws.amazon.com/cli/latest/reference/appsync/create-resolver.html) also has this option in its `--caching-config` parameter.

### New or Affected Resource(s)

* aws_appsync_resolver

### Potential Terraform Configuration

```hcl
resource "aws_appsync_resolver" "test" {
  # omitted

  caching_config {
    caching_keys = [
      "$context.identity.sub",
      "$context.arguments.id"
    ]
    ttl = 60
  }
}
```

**Comment by ackintosh** [comment]
This issue will be closed via https://github.com/terraform-providers/terraform-provider-aws/pull/12747 :octocat:

**Comment by bflad** [comment]
Support for this new configuration block has been merged and will release with version 2.62.0 of the Terraform AWS Provider, later this week. Thanks to @ackintosh for the implementation. 👍
